### PR TITLE
[MajorVersion] Project Structure, Additional Config & Require-All

### PR DESCRIPTION
**Pull request by MChambers1992**
This PR changes the file structure of the project to something more similar to other open-source projects which separates files into a more granular level. Having a separate source and test folder allows us to have the possibility of unit or integration tests in the future we could run with a test runner such as Mocha.

This PR also introduces `chatConfig.EXAMPLE.js` and `mainConfig.EXAMPLE.js` this is a solution to a problem that developers face when working on the project which is having to revert changes made in config files as they are testing before pull requests or commits. This also needed a change to be made in the .gitignore file so those files are now ignored in commits. For development of the config files the example files should be updated and developers can add the new changes to their customized config files. 

A new configuration option was added to allow developers the option to require email activated accounts on login or not, this is handy during development if quick users are needed to be added. This is a simple bool flag and an additional condition check in the login-route.

The final major change is the way `socket_modules` are added, with the use of the `require-all` package we can require all `socket_modules` and pass the `io` object to them in a single block not requiring any further consideration or bloat in the server.js file. This is a more tidy way of maintaining this file as well as removing something for developers to encounter as an issue. The package includes the option to filter certain files through regex and many other good features, a detailed breakdown can be found here; https://www.npmjs.com/package/require-all

Several files have had the code formatting tidied up moving towards.

### CODE REVIEW

**Comment by shadowoffice** [comment]
Nice work man !

File: src/server/server.js [hunk 1]
```diff
@@ -1,40 +1,38 @@
...
 
 server.listen(config.port);
-log.info('Server is on bruh and running on port: '+ config.port);
+log.info('Server is on bruh and running on port: ' + config.port);
 
 //----------------------------------
 // SET ROUTES FOR EXPRESS API HERE:
 //----------------------------------
 
+app.use(function (req, res, next) {
```
Comment (shadowoffice) [inline src/server/server.js]: Not requirt whit cors patch 👍

Comment (MChambers1992) [reply src/server/server.js]: Will update this PR at the weekend with latest changes so it's ready to merge

**Comment by MChambers1992** [comment]
Closing this pull request as it ended it being too different in scope to what is planned on the roadmap. 

The file structure I feel would be a step in the right direction and the change to have email authentication as an optional config are both changes that could be applied in separate PRs if anybody wanted to, I won't be doing that though.

**Pull request closed without merging**